### PR TITLE
add v4 images for java17

### PR DESCRIPTION
**Pull request by kaibocai**
Releasing new Host versions.
[V3 Release](PASTE_V3_LINK_HERE)
[V2 Release](PASTE_V2_LINK_HERE)

---

This PR includes:

- add v4 images for java17
- update files to include java17: 
    - core-tools-publish.yml, 
    - java-build.yml, 
    - publish-appservice-stage-sovereign-clouds.yml, 
    - publish-appservice-stage.yml, 
    - publish.yml 

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

File: host/4/bullseye/amd64/java/java17/java17-composite.template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    HOME=/home \
+    FUNCTIONS_WORKER_RUNTIME=java \
+    DOTNET_USE_POLLING_FILE_WATCHER=true \
+    HOST_VERSION=${HOST_VERSION} \
+    ASPNETCORE_CONTENTROOT=/azure-functions-host \
+    JAVA_HOME=${JAVA_HOME}
+
+# Fix from https://github.com/GoogleCloudPlatform/google-cloud-dotnet-powerpack/issues/22#issuecomment-729895157
+RUN apt-get update && \
+    apt-get install -y libc-dev wget libfreetype6 fontconfig fonts-dejavu && \
```
Comment (CooperLink) [inline host/4/bullseye/amd64/java/java17/java17-composite.template:27]: Not important for this PR but, calling out that this change may no longer be necessary.  We are apt-get ing libc-dev as a result of a GRPC issue from 2020.  Reference : https://github.com/GoogleCloudPlatform/google-cloud-dotnet-powerpack/issues/22#issuecomment-729895157

File: host/4/bullseye/amd64/java/java17/java17-slim.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    HOME=/home \
+    FUNCTIONS_WORKER_RUNTIME=java \
+    DOTNET_USE_POLLING_FILE_WATCHER=true \
+    HOST_VERSION=${HOST_VERSION} \
+    ASPNETCORE_CONTENTROOT=/azure-functions-host \
+    JAVA_HOME=${JAVA_HOME}
+
+# Fix from https://github.com/GoogleCloudPlatform/google-cloud-dotnet-powerpack/issues/22#issuecomment-729895157
+RUN apt-get update && \
+    apt-get install -y libc-dev
```
Comment (CooperLink) [inline host/4/bullseye/amd64/java/java17/java17-slim.Dockerfile:56]: Same callout as above. Not super important atm. But, we create a whole new layer here for a change that may not be required.

File: host/4/bullseye/amd64/java/java17/java17-composite.template (hunk 1 above)
Comment (kaibocai) [reply host/4/bullseye/amd64/java/java17/java17-composite.template:27]: Thank you very much for the heads up. Will validate and remove if it's unnecessary in a separate pr.

**Review by CooperLink — DISMISSED** [review]
Looks great Kaibo thank you for your contributions as always

**Review by CooperLink — APPROVED** [review]
This is great.  Thank you Kaibo.  

Kaibo removed Secure Supply Chain Violations and ensured that build test pipelines would succeed for Java17 with his changes.

**Pull request merged**